dokka-integration-tests/gradle/src/test/kotlin/AndroidComposeIT.kt

```diff
@@ -0,0 +1,158 @@
+                            dokkaWorkerLog.shouldBeAFile()
+                            dokkaWorkerLog.readText().shouldNotContainAnyOf(
+                                "[ERROR]",
+                                "[WARN]",
+                            )
+                        }
+                }
+
+                withClue("expect configurations are not resolved during configuration time") {
+                    output shouldNotContain Regex("""Configuration '.*' was resolved during configuration time""")
+                    output shouldNotContain "https://github.com/gradle/gradle/issues/2298"
+                }
+            }
+
+        withClue("expect the same HTML is generated") {
+            val expectedHtml = project.projectDir.resolve("expectedData/html")
+            val actualHtmlDir = project.projectDir.resolve("build/dokka/html")
+
+            withClue(
+                """
+                expectedHtml: ${expectedHtml.toUri()}
+                actualHtmlDir: ${actualHtmlDir.toUri()}
+                """.trimIndent()
+            ) {
+                val expectedFileTree = expectedHtml.toTreeString()
+                val actualFileTree = actualHtmlDir.toTreeString()
+                withClue((actualFileTree to expectedFileTree).sideBySide()) {
+                    expectedFileTree shouldBe actualFileTree
+
+                    actualHtmlDir shouldBeADirectoryWithSameContentAs expectedHtml
+                }
+            }
+        }
+    }
+
+    @DokkaGradlePluginTest(sourceProjectName = "it-android-compose")
+    fun `Dokka tasks should be build cacheable`(
+        project: DokkaGradleProjectRunner,
+    ) {
+        project.runner
+            .addArguments(
+                ":dokkaGenerate",
+                "--build-cache",
+            )
+            .build {
+                withClue("expect dokkaGenerate runs successfully") {
+                    shouldHaveTask(":dokkaGenerate").shouldHaveOutcome(UP_TO_DATE, SUCCESS)
+                }
+            }
+
+        project.runner
+            .addArguments(
+                ":dokkaGenerate",
+                "--build-cache",
+            )
+            .build {
+                withClue("expect dokkaGenerate tasks are loaded from cache") {
+                    shouldHaveTask(":dokkaGenerate").shouldHaveOutcome(UP_TO_DATE)
+                }
+            }
+    }
+
+    @DokkaGradlePluginTest(sourceProjectName = "it-android-compose")
+    fun `expect Dokka is compatible with Gradle Configuration Cache`(
+        project: DokkaGradleProjectRunner,
+    ) {
+        fun clearCcReports() {
+            project.file(".gradle/configuration-cache").deleteRecursively()
+            project.file("build/reports/configuration-cache").deleteRecursively()
+        }
+
+        val configCacheRunner =
+            project.runner.addArguments(
+                "clean",
+                ":dokkaGenerate",
+                "--no-build-cache",
```

Comment:
why is it necessary to specify "no build cache" here? I though configuration cache can work without knowing anything about build cache as it is more related to execution phase.

Reply:
Hmmm I suppose disabling build cache not strictly necessary, but I can imagine that a buggy Gradle task might produce some uncached output that could disrupt CC. 

TBH I can't remember why it's there. Perhaps it could be removed, but I don't think it does any harm.